conversion/resource_generation.go

```diff
@@ -0,0 +1,413 @@
+		if element == profileName {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+// If any of the resource creation fails, deletes all resources that were created
+func (r ResourceGenerationStruct) connectionProfileCleanUp(ctx context.Context, profiles []*ConnectionProfileReq) error {
+	dsClient := GetDatastreamClient(ctx)
+	for _, profile := range profiles {
+		op, err := dsClient.DeleteConnectionProfile(ctx, &datastreampb.DeleteConnectionProfileRequest{
+			Name: fmt.Sprintf("projects/%s/locations/%s/connectionProfiles/%s", profile.ConnectionProfile.ProjectId, profile.ConnectionProfile.Region, profile.ConnectionProfile.Id),
+		})
+
+		if err != nil {
+			return err
+		}
+
+		err = op.Wait(ctx)
+		if err != nil {
+			return err
+		}
+
+		if profile.ConnectionProfile.BucketName != "" {
+			gcsClient, err := storage.NewClient(ctx)
+
+			if err != nil {
+				return err
+			}
+
+			if err := gcsClient.Bucket(profile.ConnectionProfile.BucketName).Delete(ctx); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// Returns source and destination connection profiles to be created
+func (r *ResourceGenerationStruct) getResourcesForCreation(ctx context.Context, projectId string, sourceProfile profiles.SourceProfile, region string, validateOnly bool, dsClient *datastream.Client) ([]*ConnectionProfileReq, []*ConnectionProfileReq, error) {
+	var sourceProfilesToCreate []*ConnectionProfileReq
+	var dstProfilesToCreate []*ConnectionProfileReq
+
+	// Map for each region with list of all connection profiles
+	var connectionProfiles map[string][]string = make(map[string][]string)
+	var err error = nil
+
+	s := ResourceGenerationStruct{}
+	for _, profile := range sourceProfile.Config.ShardConfigurationDataflow.DataShards {
+		sourceProfileExists := false
+		dstProfileExists := false
+		if profile.SrcConnectionProfile.Name != "" {
+			// If location is not provided set it to spanner region
+			if profile.SrcConnectionProfile.Location == "" {
+				profile.SrcConnectionProfile.Location = region
+			}
+			// Check if source connection profile exists
+			sourceProfileExists, err = s.connectionProfileExists(ctx, projectId, profile.SrcConnectionProfile.Name, profile.SrcConnectionProfile.Location, connectionProfiles, dsClient, &s)
+			if err != nil {
+				return sourceProfilesToCreate, dstProfilesToCreate, err
+			}
+		}
+
+		// Destination connection profiles do not need to be validated as for their creation gcs bucket will also be created
+		if profile.DstConnectionProfile.Name != "" && !validateOnly {
+			// Check if destination connection profile exists
+			dstProfileExists, err = s.connectionProfileExists(ctx, projectId, profile.DstConnectionProfile.Name, profile.DstConnectionProfile.Location, connectionProfiles, dsClient, &s)
+			if err != nil {
+				return sourceProfilesToCreate, dstProfilesToCreate, err
+			}
+		}
+		if !sourceProfileExists {
+			id := profile.SrcConnectionProfile.Name
+			if id == "" {
+				id = "CN-" + uuid.New().String()
```

Comment:
I think we use `hb-` prefix everywhere like `hb-job-*` , `hb-pub-*`. We also don't use the full uuid if I recall correctly. Please use the naming convention used in other parts of the code.